src/helper-modules/companies-house/companies-house.service.test.ts

```diff
@@ -0,0 +1,168 @@
+      axiosError.response = {
+        data: getCompanyCompaniesHouseMalformedAuthorizationHeaderResponse,
+        status: 400,
+        statusText: 'Bad Request',
+        config: undefined,
+        headers: undefined,
+      };
+
+      when(httpServiceGet)
+        .calledWith(...expectedHttpServiceGetArguments)
+        .mockReturnValueOnce(throwError(() => axiosError));
+
+      const getCompanyPromise = service.getCompanyByRegistrationNumber(testRegistrationNumber);
+
+      await expect(getCompanyPromise).rejects.toBeInstanceOf(CompaniesHouseMalformedAuthorizationHeaderException);
+      await expect(getCompanyPromise).rejects.toThrow(`Invalid 'Authorization' header. Check that your 'Authorization' header is well-formed.`);
+      await expect(getCompanyPromise).rejects.toHaveProperty('innerError', axiosError);
+    });
+
+    it(`throws a CompaniesHouseInvalidAuthorizationException when the Companies House API returns a 401 response containing the error string 'Invalid Authorization'`, async () => {
+      const axiosError = new AxiosError();
+      axiosError.response = {
+        data: getCompanyCompaniesHouseInvalidAuthorizationResponse,
+        status: 401,
+        statusText: 'Unauthorized',
+        config: undefined,
+        headers: undefined,
+      };
+
+      when(httpServiceGet)
+        .calledWith(...expectedHttpServiceGetArguments)
+        .mockReturnValueOnce(throwError(() => axiosError));
+
+      const getCompanyPromise = service.getCompanyByRegistrationNumber(testRegistrationNumber);
+
+      await expect(getCompanyPromise).rejects.toBeInstanceOf(CompaniesHouseInvalidAuthorizationException);
+      await expect(getCompanyPromise).rejects.toThrow('Invalid authorization. Check your Companies House API key.');
+      await expect(getCompanyPromise).rejects.toHaveProperty('innerError', axiosError);
+    });
+
+    it(`throws a CompaniesHouseNotFoundException when the Companies House API returns a 404 response containing the error string 'company-profile-not-found'`, async () => {
+      const axiosError = new AxiosError();
+      axiosError.response = {
+        data: getCompanyCompaniesHouseNotFoundResponse,
+        status: 404,
+        statusText: 'Not Found',
+        config: undefined,
+        headers: undefined,
+      };
+
+      when(httpServiceGet)
+        .calledWith(...expectedHttpServiceGetArguments)
+        .mockReturnValueOnce(throwError(() => axiosError));
+
+      const getCompanyPromise = service.getCompanyByRegistrationNumber(testRegistrationNumber);
+
+      await expect(getCompanyPromise).rejects.toBeInstanceOf(CompaniesHouseNotFoundException);
+      await expect(getCompanyPromise).rejects.toThrow(`Company with registration number ${testRegistrationNumber} was not found.`);
+      await expect(getCompanyPromise).rejects.toHaveProperty('innerError', axiosError);
+    });
+
+    it('throws a CompaniesHouseException if the Companies House API returns an unknown error response', async () => {
```

Comment:
As MDM and DTFS use more specific error string rather than status code error matching, I would need to reproduce `429` and `500` errors from CH API to ascertain the error strings associated with these errors (which may be difficult), as these errors are not documented in the API documentation: https://developer-specs.company-information.service.gov.uk/companies-house-public-data-api/reference/company-profile/company-profile

As we don't specifically need to handle these errors (we just return a `500` for our endpoint for both), I don't think we need specific error handling, as in MDM we log Axios requests/responses anyway.